### ISSUE TITLE
My Devices Default View - Default set to "List" after log in.

### ISSUE THREAD

**Issue by cahsdhelpdesk** [Ylianst/MeshCentral#975]
First, from the IT director on down, the product is far above any expectation we had starting out. Many thanks!

Second, this is NOT a high priority (or really a priority in the overall scheme of things) but we would like to be able to set the "List" view as the default view under My Devices when we log in.

I am sure it is a simple change to the order in either a CSS or HTML file.
I can see the order when I view the page source but I am unable to locate the corresponding page to edit.
Can you point me to a closed incident, the file to edit or create a button to set the view as default in a future release?

Cheers,

Jeff

![List](https://user-images.githubusercontent.com/60752638/75370534-d75aba80-588a-11ea-9eb7-74d88b5ba67c.PNG)

**Comment by Ylianst** [comment]
I am back from vacation tomorrow and will take a look at this. Normally, once you select the list view in your account, it should be the default when you log back in. I will take a look at this when I get back, should be an easy fix. If you want a config.json setting that would cause all users to default to list view, let me know.

**Comment by vitko-bg** [comment]
For me it works fine - current view remains thr same after logout login. Even from different browsers.....

**Comment by cahsdhelpdesk** [comment]
Due to the environment we work in, we do not cache cookies, pages, etc.. So once we close the browser and open it back up, all those items are deleted. It would be nice to have a config.json setting so that all users default to the list view but once again, this is not a priority by any sense of the word so enjoy the rest of your vacation and your vacation from your vacation. Thanks Ylianst!

**Comment by Ylianst** [comment]
Published MeshCentral v0.4.9-o with a fix change on this. The "DeviceView", "ViewSize", "Night Mode" and "Footer Bar" settings are now server-side state. So, even without cookies or web browser state, these settings should now be kept across logins. This may be what you are asking for.

In addition, you can now put the following in the "domain" section of the config.json:

```
      "DefaultUserWebState": {
        "nightMode": 0,
        "footerBar": 1,
        "uiMode": 0,
        "deviceView": 2
      },
      "ForcedUserWebState": {
        "nightMode": 1,
        "footerBar": 1,
        "uiMode": 0,
        "deviceView": 1
      }
```

You can put one or both of these sections. The "DefaultUserWebState" are values for new users and "ForcedUserWebState" are values that are set on each page load. Possible values:

```
        "nightMode": 0/1
        "footerBar": 0/1
        "uiMode": 0/1/2
        "deviceView": 1/2/3
```

Note that the capitalization of the settings is important, so you should copy this as-is and play around with the values. Let me know if this works.

**Comment by cahsdhelpdesk** [comment]
Yes! That is what we needed. Many thanks.!
